packages/stacking/src/index.ts

```diff
@@ -0,0 +1,410 @@
+        return poxInfo.reward_cycle_length * targetBlockTime;
+      })
+  }
+
+  /**
+   * Get number of seconds until next reward cycle
+   * 
+   * @returns {Promise<number>} that resolves to a number if the operation succeeds
+   */
+  async secondsUntilNextCycle(): Promise<number> {
+    const poxInfoPromise = this.getPoxInfo();
+    const targetBlockTimePromise = await this.getTargetBlockTime();
+    const coreInfoPromise = this.getCoreInfo();
+
+    return Promise.all([poxInfoPromise, targetBlockTimePromise, coreInfoPromise])
+      .then(([poxInfo, targetBlockTime, coreInfo]) => {
+        const blocksToNextCycle =
+        (poxInfo.reward_cycle_length -
+          ((coreInfo.burn_block_height - poxInfo.first_burnchain_block_height) %
+            poxInfo.reward_cycle_length));
+        const cycleDuration = poxInfo.reward_cycle_length * targetBlockTime;
+        return blocksToNextCycle * cycleDuration;
+      })
+  } 
+
+  /**
+   * Check if stacking is enabled for next reward cycle
+   * 
+   * @returns {Promise<boolean>} that resolves to a bool if the operation succeeds
+   */
+  async stackingEnabledNextCycle(): Promise<boolean> {
+    return (await this.getPoxInfo()).rejection_votes_left_required > 0;
+  }
+
+  /**
+   * Check if account has minimum require amount of Stacks for stacking
+   * 
+   * @returns {Promise<boolean>} that resolves to a bool if the operation succeeds
+   */
+  async hasMinimumRequiredStxAmount(): Promise<boolean> {
+    const balance: BN = await this.getAccountBalance();
+    // TODO pox info should use string type instead of number
+    const min: BN = new BN((await this.getPoxInfo()).min_amount_ustx.toString());
+    return balance.gte(min);
+  }
+
+  /**
+   * Check if account can lock stx
+   * 
+   * @param {CanLockStxOptions} options - a required lock STX options object 
+   * 
+   * @returns {Promise<StackingEligibility>} that resolves to a StackingEligibility object if the operation succeeds
+   */
+  async canLockStx({
+    poxAddress, 
+    cycles
+  }: CanLockStxOptions
+  ): Promise<StackingEligibility> {
+    const balancePromise: Promise<BN> = this.getAccountBalance();
+    const poxInfoPromise = this.getPoxInfo();
+
+    return Promise.all([balancePromise, poxInfoPromise])
+      .then(([balance, poxInfo]) => {
+        const { hashMode, data } = convertBTCAddress(poxAddress);
+        const hashModeBuffer = bufferCV(new BN(hashMode, 10).toBuffer());
+        const hashbytes = bufferCV(data);
+        const poxAddressCV = tupleCV({
+          hashbytes,
+          version: hashModeBuffer,
+        });
+
+        const [contractAddress, contractName] = poxInfo.contract_id.split('.');
+
+        return callReadOnlyFunction({
+          network: this.network,
+          contractName,
+          contractAddress,
+          functionName: 'can-stack-stx',
+          senderAddress: this.address,
+          functionArgs: [
+              poxAddressCV,
+              uintCV(balance.toString()),
```

Comment:
shouldn't this be the balance - tx fees for stacking?

Reply:
I think rather than calculating this based on fees which would require another network call, this should be left up to the developer. Additionally this would not take into account sponsored transactions.

Reply:
Maybe we could double click on this one. Because the developer will likely assume that a positive response means they're good to go, which wouldn't be the case _always_. 

- Curious about the sponsored txs: are they important for stacking? why would someone pay for these tx fees?
- Could we provide an optional method like `includeTxFees` to account for this not being critical?

Reply:
It wouldn't be 100% in both cases, since fees can fluctuate and balances, minimums can change between calls. 

> Curious about the sponsored txs: are they important for stacking? why would someone pay for these tx fees?

A Stacking app or custodial service could take care of the fees for the user.